Include/Rocket/Core/StringBase.inl

```diff
@@ -330,39 +332,24 @@ void StringBase< T >::Resize(size_type new_length)
 {
 	Reserve(new_length);
 	length = new_length;
-	value[length] = '\0';
-
-	if (length == 0)
-		Clear();
+	value[length] = T();
 }
 
 // Create a lowercase version of the string
 template< typename T >
 StringBase< T > StringBase< T >::ToLower() const
 {
-	// Loop through the string, looking for an uppercase character
-	size_t copy_index = npos;
-	for (size_t i = 0; i < length; i++)
+	StringBase<T> lowercase(*this);
+	T* c = lowercase.value;
+	T* e = c + length;
```

Comment:
You might be right, but since you never said what you think it does I can't argue with you so that's not a helpful comment to me. If you tell me I might learn something.